### PR TITLE
Changes to Category Query in schema for filtering.

### PR DESCRIPTION
**Pull request by avattam06**
Added proposal for category filtering changes in CatalogGraphQl/schema.graphqls

## Problem

The current category query schema does not allow filtering with category url or name or id and  also doesn't work with multiple categories.

## Solution

The goal is to add filter on the schema to return the same data with category id, category url or category name and also able to query on multiple categories

## changes:
 // added Category filter input.
   filter: CategoryFilterInput to Query CategoryList

 type Query CategoryList(
filters: CategoryFilterInput
): [CategoryTree]

input CategoryFilterInput
{
ids: FilterEqualTypeInput,
url_key: FilterEqualTypeInput,
name: FilterMatchTypeInput
} 

## Requested Reviewers
@paliarush 
@akaplya

### CODE REVIEW

**Comment by magento-cicd2** [comment]
[CLA assistant check](https://cla.dev.magento.com/magento/architecture?pullRequest=294) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dev.magento.com/magento/architecture?pullRequest=294) before we can accept your contribution.<br/>

File: design-documents/graph-ql/coverage/CategoryFiltering.graphqls [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+   filter: CategoryFilterInput @doc(description: "Identifies which Category filter inputs to search for and return.")
+   ): CategoryTree
+    @resolver(class: "Magento\\CatalogGraphQl\\Model\\Resolver\\CategoryTree") @doc(description: "The category query searches for categories that match the criteria specified in the search and filter attributes.") @cache(cacheIdentity: "Magento\\CatalogGraphQl\\Model\\Resolver\\Category\\CategoryTreeIdentity")
+}
+
+# added CategoryFilterInput with id, url_key and name as input filters.
+
+input CategoryFilterInput @doc(description: "CategoryFilterInput defines the filters to be used in the search. A filter contains at least one attribute, a comparison operator, and the value that is being searched for.") {
+    id: FilterEqualTypeInput @doc(description: "Filter by category ID that uniquely identifies and filters the category."),
+    url_key: FilterEqualTypeInput @doc(description: "Filter by the part of the URL that identifies the category"),
```
Comment (akaplya) [inline design-documents/graph-ql/coverage/CategoryFiltering.graphqls]: URL key and name should support search by incomplete value.

Comment (avattam06) [reply design-documents/graph-ql/coverage/CategoryFiltering.graphqls]: added new changes for the schema.

**Comment by akaplya** [comment]
Approved

**Pull request merged**